### ISSUE TITLE
The bridge is being rewritten in Go

### ISSUE THREAD

**Issue by tulir** [mautrix/signal#372]
The Python version will likely not receive any further updates. Once the Go bridge is ready for general use, it will be merged into this repository and replace the Python version. The new bridge will likely support a direct upgrade from the last old version (v0.4.3) preserving all portal rooms. However, Signal logins are currently managed by signald, which the new bridge will not use, so those will not be preserved and all users will have to re-link the bridge.

**Comment by NilsIrl** [comment]
Thinking of having a go/look at #1. In light of this, what are things to consider?

**Comment by mattcen** [comment]
Is the plan to link against [libsignal[(https://github.com/signalapp/libsignal) (I have no idea how easy that is since it's written in Rust), or are you writing from scratch?

**Comment by maltee1** [comment]
Against libsignal using an FFI layer

**Comment by mietzen** [comment]
Is there a roadmap for the golang rewrite? Will this be pushed into this repo?

**Comment by Dual-0** [comment]
> Is there a roadmap for the golang rewrite? Will this be pushed into this repo?

read the [repo](https://github.com/mautrix/signalgo)  description:
> Go rewrite of mautrix-signal - This is a temporary project that will be merged into the main mautrix-signal repo when ready

**Comment by pbek** [comment]
> Is there a roadmap for the golang rewrite? Will this be pushed into this repo?

It was pushed today. But I had issues, see #382.

**Comment by Tropaion** [comment]
Sadly not working for me, bot doesn't respond to anything.
I had the previous version installed with pip in an venv.
I copied conig.yaml and registration.yaml, deleted the venv and installed the go version with systemd service.
I can start to bridge and I didn't recognized any error either, but I don't get any responsed.
When I write a message, it clearly sees it because it says:
`Feb 17 01:51:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:51:55+01:00 DBG Decrypted olm message component=crypto olm_session_id=hO6srLgiq5ylSHp5R7g+mZjiF8F+tPyP/TiqA+biH8k sender=@fabian:flowerhouse.at trace_id=01:51:55.160123 type=m.room.encrypted
Feb 17 01:51:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:51:55+01:00 DBG Received inbound group session component=crypto is_scheduled=false max_age=0s max_messages=0 sender=@fabian:flowerhouse.at sender_key=jXnd/sDu9Mjy/qO0V79tVM/oSRb4InwF1fd9h7CDg1M session_id=rfFmneUZzT1LnJ5CE2x7z4lfeptSxT4C14wa2JFUEE8 trace_id=01:51:55.160123 type=m.room.encrypted
Feb 17 01:52:25 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:52:25+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30013.32983 method=GET req_id=18 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150118_23155148_20_521007_34987_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:52:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:52:55+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30009.236038 method=GET req_id=19 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150118_23155148_20_521007_34987_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:53:25 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:53:25+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30007.945636 method=GET req_id=20 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150121_23155175_24_521010_34990_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:53:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:53:55+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30008.163629 method=GET req_id=21 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155199_26_521011_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:54:25 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:54:25+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30008.402647 method=GET req_id=22 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155211_26_521011_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:54:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:54:55+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30007.260044 method=GET req_id=23 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155238_26_521011_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:55:25 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:55:25+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30007.305478 method=GET req_id=24 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155250_26_521011_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:55:55 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:55:55+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30006.617756 method=GET req_id=25 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155279_26_521013_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
Feb 17 01:56:25 ChatMatrix mautrix-signal[253539]: 2024-02-17T01:56:25+01:00 DBG Request completed as_user_id=@signalbot:flowerhouse.at duration=30006.677926 method=GET req_id=26 response_length=219 response_mime=application/json status_code=200 url=http://localhost:8008/_matrix/client/v3/sync?filter=1&since=s150122_23155289_26_521015_34992_28_26889_2246911_0_2&timeout=30000&user_id=%40signalbot%3Aflowerhouse.at
`
